src/main/kotlin/platform/architectury/creator/ArchitecturyProjectSettingsWizard.kt

```diff
@@ -0,0 +1,318 @@
+        CoroutineScope(Dispatchers.Swing).launch {
+            loadingBar.isIndeterminate = true
+            loadingBar.isVisible = true
+
+            updateForm()
+
+            loadingBar.isIndeterminate = false
+            loadingBar.isVisible = false
+        }
+    }
+
+    override fun getComponent(): JComponent {
+        return panel
+    }
+
+    override fun updateStep() {
+        val (conf) = modUpdateStep<ArchitecturyProjectConfig>(creator, modNameField) ?: return
+        config = conf
+
+        if (creator.configs.indexOf(conf) != 0) {
+            modNameField.isEditable = false
+        }
+
+        title.icon = PlatformAssets.ARCHITECTURY_ICON_2X
+        title.text = "<html><font size=\"5\">Architectury Settings</font></html>"
+
+        minecraftVersionLabel.text = "Minecraft Version"
+
+        licenseBox.model = EnumComboBoxModel(License::class.java)
+        licenseBox.selectedItem = License.ALL_RIGHTS_RESERVED
+
+        if (versions != null || currentJob?.isActive == true) {
+            return
+        }
+        currentJob = updateVersions()
+    }
+
+    private fun setForgeVersion(data: Data) {
+        forgeVersionBox.model = CollectionComboBoxModel(data.forgeVersions.subList(0, min(50, data.forgeVersions.size)))
+        forgeVersionBox.selectedIndex = data.forgeSelectedIndex
+    }
+
+    private fun setFabricVersion(data: Data) {
+        fabricVersionBox.model = CollectionComboBoxModel(
+            data.fabricVersions.subList(0, min(50, data.fabricVersions.size))
+        )
+        fabricVersionBox.selectedIndex = data.fabricSelectedIndex
+    }
+
+    private fun setFabricApiVersion(data: Data) {
+        fabricApiVersionBox.model = CollectionComboBoxModel(
+            data.fabricApiVersions.subList(0, min(50, data.fabricApiVersions.size))
+        )
+        fabricApiVersionBox.selectedIndex = data.fabricApiSelectedIndex
+    }
+
+    private fun setArchitecturyApiVersion(data: Data) {
+        architecturyApiVersionBox.model = CollectionComboBoxModel(
+            data.architecturyApiVersions.subList(0, min(50, data.architecturyApiVersions.size))
+        )
+        architecturyApiVersionBox.selectedIndex = data.architecturyApiSelectedIndex
+    }
+
+    private val version: SemanticVersion?
+        get() = minecraftVersionBox.selectedItem as? SemanticVersion
+
+    override fun validate(): Boolean {
+        return super.validate() && !loadingBar.isVisible
+    }
+
+    override fun isStepVisible(): Boolean {
+        return creator.configs.any { it is ArchitecturyProjectConfig }
+    }
+
+    override fun onStepLeaving() {
+        currentJob?.cancel()
+    }
+
+    override fun updateDataModel() {
+        val conf = this.config ?: return
+
+        conf.pluginName = this.modNameField.text
+        conf.description = this.descriptionField.text
+        conf.website = this.websiteField.text
+
+        conf.modRepo = this.repositoryField.text
+        conf.modRepo = this.issueField.text
```

Comment:
Think you're setting the wrong field here

Reply:
oop